shared/editor/plugins/Prism.ts

```diff
@@ -86,13 +90,25 @@ function getDecorations({
     });
   }
 
-  blocks.forEach((block) => {
+  blocks.forEach(async (block) => {
     let startPos = block.pos + 1;
     const language = block.node.attrs.language;
-    if (!language || language === "none" || !refractor.registered(language)) {
+    const isExcluded = EXCLUDED_LANGUAGES.includes(language);
+    if (!language || language === "none" || isExcluded) {
       return;
     }
 
+    if (!refractor.registered(language)) {
+      try {
+        // Dynamically load and register the language module
+        const mod = await import(`refractor/lang/${language}`);
```

Comment:
It might be better to leave this method as non `async` and actually use a standard then/catch here.

Two things left to consider if you're willing:

- This loop gets triggered a lot as the user is typing, we need to record that we're already loading the module and avoid doing so again, probably in a local map/object.
- When the module loads we can then dispatch a transaction with some metadata that tells the editor to re-render, this should solve the issue in the screencast. You can see how we listen to some metadata further down on L184 to trigger a render.

Reply:
Still getting used to the codebase and architecture, these suggestions surely help me understanding it better. I'll give it a shot and try to implement these changes. Thanks! 🙏🏽